src/main/kotlin/com/github/gradle/node/NodePlugin.kt

```diff
@@ -25,8 +25,10 @@ class NodePlugin : Plugin<Project> {
         val nodeExtension = NodeExtension.create(project)
         addGlobalTypes()
         addTasks()
-        addNpmRule()
-        addYarnRule()
+        if (!nodeExtension.disableTaskRules.get()) {
```

Comment:
I think we can guarantee that this works if we move this check into the rule itself, e.g. only create the task if `disableTaskRules` is `false`